### PR TITLE
Add Grafana and Loki to demo

### PR DESCRIPTION
**Pull request by pellared**
## Why

Fixes https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1388

Doing the best to address https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1708#discussion_r1040446795

## What

1. Add Grafana and Loki for better UX
1. The client makes a call in a loop and the automated test is polling instead of one single hardcoded sleep

## Tests

CI and local

image

### CODE REVIEW

File: examples/demo/Client/Program.cs [hunk 1]
```diff
@@ -27,8 +27,12 @@
...
-using var httpClient = new HttpClient();
-var content = await httpClient.GetStringAsync(uri);
-Console.WriteLine(content);
-return 0;
+while (true)
+{
+    Thread.Sleep(5000);
+
+    var uri = args[0];
+    using var httpClient = new HttpClient();
```
Comment (Kielek) [inline examples/demo/Client/Program.cs:35]: https://www.aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

Comment (pellared) [reply examples/demo/Client/Program.cs]: Thanks 👍 
Addressed https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1882/commits/9dc874530fd11c91111146dcc429334af901309d

**Review by Kielek — APPROVED** [review]

**Comment by pellared** [comment]
@open-telemetry/dotnet-instrumentation-maintainers PTAL

**Review by RassK — APPROVED** [review]

File: examples/demo/Client/Program.cs [hunk 2]
```diff
@@ -27,8 +27,11 @@
...
 
-var uri = args[0];
+var url = args[0];
 using var httpClient = new HttpClient();
-var content = await httpClient.GetStringAsync(uri);
-Console.WriteLine(content);
-return 0;
+while (true)
+{
+    Thread.Sleep(5000);
```
Comment (pjanotti) [inline examples/demo/Client/Program.cs:34]: Why not `await Task.Delay`?

**Review by pjanotti — APPROVED** [review]

**Pull request merged**